### PR TITLE
Tests for functions in HelpersTest.kt

### PR DESCRIPTION
**Pull request by SunCatMC**
## Purpose
Adding tests for functions in HelpersTest.kt

### CODE REVIEW

**Comment by SunCatMC** [comment]
There's a questionable test inside a comment block, aside from that it is done

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+class HelpersTest : StringSpec({
+    "toFailedModuleEither when given Throwable in Either should return FailedModuleResponse with that 1 throwable" {
+        val moduleEither = throwable.left().toFailedModuleEither()
+
+        moduleEither.shouldBeLeft()
+        (moduleEither as Either.Left).a.shouldBe(FailedModuleResponse(listOf(throwable)))
+    }
+
+    /* commented out as I don't know yet whether this is a feature or not
+    "toFailedModuleEither when given success in Either should return the success in ModuleResponse" {
```
Comment (urielsalis) [inline src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:28]: This should probably exist :P

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
+
+        moduleEither.shouldBeRight()
+        (moduleEither as Either.Right).b.shouldBe(ModuleResponse)
+    }
+
+    "assertAfter when given 2 times first of which is before the second should return OperationNotNeededModuleResponse in Either" {
+        val moduleEither = assertAfter(A_SECOND_AGO, NOW)
+
+        moduleEither.shouldBeLeft()
+        (moduleEither as Either.Left).a.shouldBe(OperationNotNeededModuleResponse)
```
Comment (urielsalis) [inline src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:296]: moduleEither shouldBeLeft OperationNotNeededModuleResponse

same with Right :P, that replaces the .shouldBeLeft() and the casting for shouldBe

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt (hunk 1 above)
Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:28]: k

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt (hunk 2 above)
Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:296]: ah ok

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt (hunk 1 above)
Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:26]: done in https://github.com/mojira/arisa-kt/pull/433/commits/7c5bddd90ca644a8f09e26c293a336d735ceb72f

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt (hunk 2 above)
Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt]: done in https://github.com/mojira/arisa-kt/pull/433/commits/7c5bddd90ca644a8f09e26c293a336d735ceb72f

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt [hunk 3]
```diff
@@ -0,0 +1,268 @@
...
+    "assertEmpty when given empty list should return ModuleResponse in Either" {
+        val moduleEither = assertEmpty(listOf<Int>())
+
+        moduleEither.shouldBeRight(ModuleResponse)
+    }
+
+    "assertEmpty when given nonempty list should return OperationNotNeededModuleResponse in Either" {
+        val moduleEither = assertEmpty(listOf<Int>(1337))
+
+        moduleEither.shouldBeLeft(OperationNotNeededModuleResponse)
```
Comment (urielsalis) [inline src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:93]: super minor but you can use spaces and its more beautiful xd


moduleEither.shouldBeLeft(OperationNotNeededModuleResponse) is the same as 
        moduleEither shouldBeLeft OperationNotNeededModuleResponse

Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:93]: ahhhh ok

Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:93]: is it kotest only? or how do you know it if works for others?

Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt]: fixed in https://github.com/mojira/arisa-kt/pull/433/commits/c97ce8396473fd6a1d55d8cd40224dfd4c095dec

**Review by urielsalis — APPROVED** [review]

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt [hunk 4]
```diff
@@ -0,0 +1,267 @@
...
+        }
+        fun successful2(): Either<Throwable, Unit> {
+            checker += "b"
+            return Unit.right()
+        }
+        fun successful3(): Either<Throwable, Unit> {
+            checker += "c"
+            return Unit.right()
+        }
+        val moduleEither = tryRunAll(listOf({successful1()}, {successful2()}, {throwable.left()},
```
Comment (chandler05) [inline src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:177]:
```suggestion
        val moduleEither = tryRunAll(listOf({ successful1() }, { successful2() }, { throwable.left() },
```
More formatting

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt [hunk 5]
```diff
@@ -0,0 +1,267 @@
...
+            {Throwable(message="different error").left()}, {successful3()}))
```
Comment (chandler05) [inline src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt:178]:
```suggestion
            { Throwable(message = "different error").left() }, { successful3() }))
```
Last formatting error

**Review by chandler05 — COMMENTED** [review]
Just a few formatting errors that are causing the PR to fail

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt [hunk 6]
```diff
@@ -0,0 +1,267 @@
...
+        }
+        fun successful2(): Either<Throwable, Unit> {
+            checker += "b"
+            return Unit.right()
+        }
+        fun successful3(): Either<Throwable, Unit> {
+            checker += "c"
+            return Unit.right()
+        }
+        val moduleEither = tryRunAll(listOf({successful1()}, {successful2()}, {successful3()}))
```
[the remark this answers is not in the document]
Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt]: fixed by https://github.com/mojira/arisa-kt/pull/433/commits/198d97a8e7f4326a4a5fa11a8e6ae7b5988c07f2

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt (hunk 4 above)
Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt]: fixed by https://github.com/mojira/arisa-kt/pull/433/commits/198d97a8e7f4326a4a5fa11a8e6ae7b5988c07f2

File: src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt (hunk 5 above)
Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt]: fixed by https://github.com/mojira/arisa-kt/pull/433/commits/198d97a8e7f4326a4a5fa11a8e6ae7b5988c07f2

Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt]: ah there's a bit more k

Comment (SunCatMC) [reply src/test/kotlin/io/github/mojira/arisa/modules/HelpersTest.kt]: fixed by https://github.com/mojira/arisa-kt/pull/433/commits/4dabc044716fe33694ff7843f0107a7e543c883f

**Review by urielsalis — APPROVED** [review]

**Pull request merged**